jmh/src/main/java/com/uber/nullaway/jmh/NullawayJavac.java

```diff
@@ -0,0 +1,158 @@
+    } else {
+      int dot = version.indexOf(".");
+      if (dot != -1) {
+        version = version.substring(0, dot);
+      }
+    }
+    return Integer.parseInt(version);
+  }
+
+  public static void main(String[] args) throws Exception {
+    // Check at runtime for appropriate Java version.  For convenience we keep the code building
+    // with Java 8.
+    if (getJavaRuntimeVersion() < 11) {
+      throw new RuntimeException(
+          "Must be run on JDK 11 or greater; version is " + System.getProperty("java.version"));
+    }
+    NullawayJavac nb = new NullawayJavac();
+    nb.prepareForSimpleTest();
+    for (int i = 0; i < 5; i++) {
+      System.out.println(i);
+      nb.testCompile();
+    }
+    System.out.println("SUCCESS");
+  }
+
+  private List<JavaFileObject> compilationUnits;
+  private JavaCompiler compiler;
+  private DiagnosticListener<JavaFileObject> diagnosticListener;
+  private StandardJavaFileManager fileManager;
+  private List<String> options;
+
+  public void prepareForSimpleTest() throws IOException {
+    String testClass =
+        "package com.uber;\n"
+            + "import java.util.*;\n"
+            + "class Test {   \n"
+            + "  public static void main(String args[]) {\n"
+            + "    Set<Short> s = null;\n"
+            + "    for (short i = 0; i < 100; i++) {\n"
+            + "      s.add(i);\n"
+            + "      s.remove(i - 1);\n"
+            + "    }\n"
+            + "    System.out.println(s.size());"
+            + "  }\n"
+            + "}\n";
+    compilationUnits = Collections.singletonList(new JavaSourceFromString("Test", testClass));
+    finishSetup("com.uber");
+  }
+
+  public void prepare(List<String> sourceFileNames, String annotatedPackages) throws IOException {
+    compilationUnits = new ArrayList<>();
+    for (String sourceFileName : sourceFileNames) {
+      String content = readFile(sourceFileName);
+      String classname =
+          sourceFileName.substring(
+              sourceFileName.lastIndexOf(File.separatorChar) + 1, sourceFileName.indexOf(".java"));
+      compilationUnits.add(new JavaSourceFromString(classname, content));
+    }
+
+    finishSetup(annotatedPackages);
+  }
+
+  /**
+   * Finishes setup of options and state for running javac, assuming that {@link #compilationUnits}
+   * has already been set up.
+   *
+   * <p>To pass the appropriate {@code -processorpath} argument to the spawned javac, we make this
+   * project depend on NullAway and Error Prone Core, and then pass our own classpath as the
+   * processorpath. Note that this makes (dependencies of) NullAway and Error Prone visible on the
+   * <emph>classpath</emph> for the spawned javac instance as well. So, if a benchmark depends on
+   * some library that NullAway depends on (e.g., Guava), the dependence will be magically
+   * satisfied. Note that this could lead to problems for benchmarks that depend on a conflicting
+   * version of a library.
```

Comment:
Good to know. I assume there was no easy way of configuring the `processorpath` for the spawned `javac` in Gradle or otherwise pass that info. Right? (I think this should work for the cases we care about)

Reply:
So now that I think about it, we could possibly do this in a slightly cleaner fashion by sticking the processor path in a JVM property (like how we pass the path to the source files).  It'll take some thinking but might be worth it.  Let me try.

Reply:
I tinkered and did find a way to run NullAway without making the project depend on it, but it  had a flaw.  Without including NullAway as a dep of the project, its classes get reloaded each time the compiler runs, which really messes up performance measurement.  So we need to do things like this.  I've updated the comments.